path-manager/app/controllers/PathManagerController.scala

```diff
@@ -28,37 +28,55 @@ object PathManagerController extends Controller {
   }
 
   def registerExistingPath(id: Long) = Action { request =>
-    request.body.asJson.map(_.as[PathRecord]).map { submission =>
-      if (id != submission.identifier) {
+    val maybeSubmission = request.body.asJson.map(_.as[PathRecord])
+    maybeSubmission match {
+      case (Some(path)) if path.identifier != id => {
         PathOperationErrors.increment
         Logger.warn("registerExistingPath failed, identifier in url and body do not match")
         BadRequest("identifier in url and body do not match")
-      } else if (submission.`type` != "canonical") {
-        PathOperationErrors.increment
-        Logger.warn("registerExistingPath failed, only canonical paths can be updated at present")
-        BadRequest("only canonical paths can be updated at present")
-      } else {
-        PathStore.register(submission) match {
+      }
+
+      case Some(path) if path.`type` == "alias" => {
+        PathStore.registerAlias(path) match {
+          case Left(error) => BadRequest(error)
+          case Right(records) => {
+            PathMigrationRegistrations.increment
+            argoOk(Json.toJson(records))
+          }
+        }
+      }
+      case Some(path) if path.`type` == "canonical"  => {
+        PathStore.registerCanonical(path) match {
           case Left(error) => BadRequest(error)
           case Right(records) => {
             PathMigrationRegistrations.increment
             argoOk(Json.toJson(records))
           }
         }
       }
-    } getOrElse{
-      PathOperationErrors.increment
-      Logger.warn("registerExistingPath failed, unable to parse PathRecord from request body")
-      BadRequest("unable to parse PathRecord from request body")
+      case Some(_)  => {
+        PathOperationErrors.increment
+        Logger.warn("registerExistingPath failed, only canonical and alias paths can be updated at present")
+        BadRequest("only canonical paths can be updated at present")
```

Comment:
should this BadRequest message mention aliases (like the log line)

Reply:
Absolutely should.